host/internal/server.go

```diff
@@ -251,6 +256,9 @@ func (h *sessionHandler) HandleSession(sess gssh.Session) {
 		}, func(err error) {
 			cancel()
 		})
+	} else {
+		// write to client to notify them that they have connected to a read-only session
+		io.WriteString(sess, "\r\n=== Attached to read-only session ===\r\n\r\n")
```

Comment:
Mind adding a ftest for this? You may refer to https://github.com/jingweno/upterm/blob/master/ftests/client_test.go#L77 as an example and test that the client won't be able to insert any input.